lib/services/serviceBus/lib/notificationhubservice.js

```diff
@@ -290,14 +290,14 @@ NotificationHubService.prototype.updateRegistration = function (registration, op
       delete registration.Expiry;
   }
 
-  if (registration.RegistrationId != null) {
+  if (registration.RegistrationId !== null || registration.RegistrationId !== undefined) {
```

Comment:
Actually I played around with the delete operator to find out what happens when you try to delete a property from an object, when the property does not exist in the object. It turns out, it is completely fine  as you can see below.

```bash
node
> obj = {}
{}
> delete obj.foo
true
> obj.foo = null
null
> obj
{ foo: null }
> delete obj.foo
true
> obj.foo = undefined
undefined
> obj
{ foo: undefined }
> delete obj.foo
true
```
However, one needs to make sure that the object is actually defined before executing `delete`
```bash
> delete baz.foo
ReferenceError: baz is not defined
    at repl:1:1
    at ContextifyScript.Script.runInThisContext (vm.js:50:33)
    at REPLServer.defaultEval (repl.js:240:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:441:10)
    at emitOne (events.js:121:20)
    at REPLServer.emit (events.js:211:7)
    at REPLServer.Interface._onLine (readline.js:282:10)
    at REPLServer.Interface._line (readline.js:631:8)
```

So, I think we should simply do the following:

```javascript
if (registration) delete registration.registrationId;
```